overlord/managers_test.go

```diff
@@ -428,3 +413,86 @@ type: kernel`
 		"snappy_mode":   "try",
 	})
 }
+
+func (ms *mgrsSuite) installLocalTestSnap(c *C, snapYamlContent string) *snap.Info {
+	st := ms.o.State()
+
+	snapPath := makeTestSnap(c, snapYamlContent)
+	snapf, err := snap.Open(snapPath)
+	c.Assert(err, IsNil)
+	info, err := snap.ReadInfoFromSnapFile(snapf, nil)
+	c.Assert(err, IsNil)
+
+	// store current state
+	snapName := info.Name()
+	var snapst snapstate.SnapState
+	snapstate.Get(st, snapName, &snapst)
+
+	ts, err := snapstate.InstallPath(st, snapName, snapPath, "", snapstate.DevMode)
+	c.Assert(err, IsNil)
+	chg := st.NewChange("install-snap", "...")
+	chg.AddAll(ts)
+
+	st.Unlock()
+	err = ms.o.Settle()
+	st.Lock()
+	c.Assert(err, IsNil)
+
+	c.Assert(chg.Status(), Equals, state.DoneStatus, Commentf("install-snap change failed with: %v", chg.Err()))
+
+	// ensure its different from before
+	var newSnapst snapstate.SnapState
+	snapstate.Get(st, snapName, &newSnapst)
+	c.Assert(newSnapst.CurrentSideInfo().Revision.Unset(), Equals, false)
+	c.Assert(snapst.CurrentSideInfo(), Not(DeepEquals), newSnapst.CurrentSideInfo())
+
+	return info
+}
+
+func (ms *mgrsSuite) TestHappyRevert(c *C) {
+	st := ms.o.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snapYamlContent := `name: foo
+apps:
+ bar:
+  command: bin/bar
+`
+	ms.installLocalTestSnap(c, snapYamlContent+"version: 1.0")
+	ms.installLocalTestSnap(c, snapYamlContent+"version: 2.0")
+
+	// ensure we are on r2
+	c.Assert(osutil.FileExists(filepath.Join(dirs.SnapBlobDir, "foo_x2.snap")), Equals, true)
+	mup := systemd.MountUnitPath("/snap/foo/x2", "mount")
+	content, err := ioutil.ReadFile(mup)
+	c.Assert(err, IsNil)
+	c.Assert(string(content), Matches, "(?ms).*^Where=/snap/foo/x2")
+	c.Assert(string(content), Matches, "(?ms).*^What=/var/lib/snapd/snaps/foo_x2.snap")
+
+	// now do the revert
+	ts, err := snapstate.Revert(st, "foo")
+	c.Assert(err, IsNil)
+	chg := st.NewChange("revert-snap", "...")
+	chg.AddAll(ts)
+
+	st.Unlock()
+	err = ms.o.Settle()
+	st.Lock()
+	c.Assert(err, IsNil)
+
+	c.Assert(chg.Status(), Equals, state.DoneStatus, Commentf("revert-snap change failed with: %v", chg.Err()))
+
+	// ensure we are back to x1 after the revert
+	c.Assert(osutil.FileExists(filepath.Join(dirs.SnapBlobDir, "foo_x1.snap")), Equals, true)
```

Comment:
I don't understand this test, is checking only things that are true independently of what is active (mount units, snap blob), which is ok, but it should check also something that changes like wrappers (I think some other tests here do that),

atm I can comment out the revert invocation and it passes anyway